src/main/scala/io/applicative/datastore/DatastoreService.scala

```diff
@@ -42,6 +42,15 @@ object DatastoreService extends Datastore with ReflectionHelper {
     Key(cloudDataStore.allocateId(incompleteKey))
   }
 
+  override def add[E <: BaseEntity : TypeTag](entity: E): Future[E] = Future {
```

Comment:
Just noticed you're importing `scala.concurrent.ExecutionContext.Implicits.global` for Future executions - its better to take an implicit context as a parameter, that way the decision on which context to use can be made upstream by the user of the library.